R/tags.R

```diff
@@ -1,6 +1,31 @@
 #' @import utils digest
 NULL
 
+# Like base::paste, but converts all string args to UTF-8 first.
+paste8 <- function(..., sep = " ", collapse = NULL) {
+  args <- c(
+    lapply(list(...), enc2utf8),
+    list(
+      sep = if (is.null(sep)) sep else enc2utf8(sep),
+      collapse = if (is.null(collapse)) collapse else enc2utf8(collapse)
+    )
+  )
+
+  # The easy thing would be to call `do.call("paste", args)`, but according to
+  # http://rpubs.com/hadley/do-call2 this will inline the values, which could be
+  # potentially large strings.
+  #
+  # These lines are basically the equivalent of (for a single unnamed arg):
+  # paste(args[[1L]], sep = args[["sep"]], collapse = args[["collapse"]])
+  argNames <- ifelse(nzchar(names(args)), as.list(names(args)), 1:length(args))
+  argz <- lapply(argNames, function(x) {
+    eval(substitute(quote(args[[x]]), list(x=x)))
+  })
+  names(argz) <- names(args)
```

Comment:
Yes, they're total overkill.  Inlining the values is actually a slight performance _enhancement_ most of the time, because it avoids a single variable-value lookup. The downside only comes if an exception is thrown, because it takes much longer to generate the traceback.